client/src/components/RecipeList/components/RecipeCard.jsx

```diff
@@ -1,42 +1,195 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import { Card } from 'antd';
+import {
+  Card, Modal, Button, Tag, Descriptions, Divider,
+} from 'antd';
+import {
+  InfoCircleOutlined, StarOutlined, ExportOutlined, CloseCircleOutlined, StarFilled,
+} from '@ant-design/icons';
+import RecipeNames from './RecipeConsts';
 
 const { Meta } = Card;
 
-const openRecipe = (link) => {
-  window.open(link, 'noopener noreferrer');
-};
+class RecipeCard extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { showModal: false, viewMore: false };
+    this.showModal = this.showModal.bind(this);
+    this.hideModal = this.hideModal.bind(this);
```

Comment:
this is unnecessary, because when you declare showModal and hideModal within the component you actually declare this.showModal. Just initialize the state in the constructor

Reply:
There is only one problem - I changed the showModal and hideModal functions from arrow notation to function notation, because eslint wouldn't let me define them otherwise (It would say "unexpected token =" like I showed you) - If I don't bind the function to this in the constructor, the buttons don't work.